src/compile/data.ts

```diff
@@ -120,18 +117,22 @@ export function parseFacetData(model: FacetModel): DataComponent {
 
 export function parseLayerData(model: LayerModel): DataComponent {
   return {
-    source: source.parseLayer(model),
+    // filter and formatParse could cause us to not be able to merge into parent
```

Comment:
What's the rationale for reordering here?   

In any case, we should order this consistently for facet and unit for readability.